source/write/bulk-write.txt

```diff
@@ -0,0 +1,298 @@
+   :end-before: end-bulk-update-many
+   :language: c
+   :dedent:
+
+.. _c-bulk-add-replace:
+
+Replace Operations
+~~~~~~~~~~~~~~~~~~
+
+A replace operation removes all fields and values of a specified document and
+replaces them with new ones. To perform a replace operation, add the replacement
+instructions to the ``mongoc_bulk_operation_t``, which queues the operation as part
+of the bulk write.
+
+The following example calls the ``mongoc_bulk_operation_replace_one_with_opts()`` function,
+passing a query filter, replacement document, and the ``mongoc_bulk_operation_t`` value as
+parameters:
+
+.. literalinclude:: /includes/write/bulk-write.c
+   :start-after: start-bulk-replace-one
+   :end-before: end-bulk-replace-one
+   :language: c
+   :dedent:
+
+To replace multiple documents, call ``mongoc_bulk_operation_replace_one_with_opts()``
+for each document.
+
+.. _c-bulk-add-delete:
+
+Delete Operations
+~~~~~~~~~~~~~~~~~
+
+To perform a delete operation, add the delete instructions to the ``mongoc_bulk_operation_t``,
+which queues the operation as part of the bulk write.
+
+The following example calls the ``mongoc_bulk_operation_remove_one_with_opts()`` function,
+passing a query filter and the ``mongoc_bulk_operation_t`` value as parameters:
+
+.. literalinclude:: /includes/write/bulk-write.c
+   :start-after: start-bulk-delete-one
+   :end-before: end-bulk-delete-one
+   :language: c
+   :dedent:
+
+To delete multiple documents, call the ``mongoc_bulk_operation_remove_many_with_opts()``
+function and pass in the same parameters. This instructs the driver to delete *all* documents
+that match your query filter.
+
+The following example queues a delete many operation to the bulk write:
+
+.. literalinclude:: /includes/write/bulk-write.c
+   :start-after: start-bulk-delete-many
+   :end-before: end-bulk-delete-many
+   :language: c
+   :dedent:
+
+Run the Bulk Operation
+----------------------
+
+To run each write operation queued in the bulk write, call the ``mongoc_bulk_operation_execute()``
+function. This function accepts the following parameters:
+
+- **mongoc_bulk_operation_t value**: Contains the instructions for each write operation
+
+- **Results location**: Specifies a pointer to overwritable storage that will contain
+  operation results, or ``NULL``
+
+- **Error location**: Specifies a location for an error value, or ``NULL``
+
+The following example performs the :ref:`insert <c-bulk-add-insert>`,
+:ref:`update <c-bulk-add-update>`, :ref:`replace <c-bulk-add-replace>`,
+and :ref:`delete <c-bulk-add-delete>` operations specified in the preceding
+sections of this guide by calling the ``mongoc_bulk_operation_execute()``
+function:
+
+.. literalinclude:: /includes/write/bulk-write.c
+   :start-after: start-bulk-run
+   :end-before: end-bulk-run
+   :language: c
+   :dedent:
+
+If any of the write operations fail, the {+driver-short+} raises a
+``mongoc_bulkwriteexception_t`` error and does not perform any further operations.
```

Comment:
Remove mention of `mongoc_bulkwriteexception_t`. `mongoc_bulkwriteexception_t` applies to the [mongoc_bulkwrite_t](https://mongoc.org/libmongoc/current/mongoc_bulkwrite_t.html). `mongoc_bulkwrite_t` requires server 8.0. `mongoc_bulk_operation_t` can be used with any supported server version.

```suggestion
If any of the write operations fail, the {+driver-short+} sets the output error and does not perform any further operations.
```